boss-ws/src/main/java/org/genomebridge/boss/http/objectstore/GCSObjectStore.java

```diff
@@ -111,6 +114,37 @@ private PrivateKey getKey() throws Exception {
         return mKey;
     }
 
+    @Override
+    public URI generateResumableUploadURL(String objectName) {
+        try{
+            URI uri = generateResumableURI(StringUtils.deleteWhitespace(objectName));
+            HttpURLConnection conn = (HttpURLConnection) uri.toURL().openConnection();
+            conn.setDoOutput(true);
+            conn.setDoInput(true);
+            conn.setUseCaches(false);
+            conn.setRequestMethod(HttpMethod.PUT);
+            conn.setRequestProperty("x-goog-resumable", "start");
+            conn.getOutputStream().close();
+            if (conn.getResponseCode() == Response.Status.CREATED.getStatusCode()) {
+                return URI.create(conn.getHeaderField("Location"));
+            }else{
+                throw new ObjectStoreException("Can't get resumable URL. "+conn.getResponseMessage());
```

Comment:
The else is unnecessary -- line above is a return.  This is minor.  More importantly, this throw is going to be caught by the catch on the next line, and you'll have two copies of "Can't get resumable URL." in the error message.